src/subcommands/deploy.ts

```diff
@@ -0,0 +1,306 @@
+// Copyright 2021 Deno Land Inc. All rights reserved. MIT license.
+
+import { fromFileUrl, join, normalize, Spinner, wait } from "../../deps.ts";
+import { error } from "../error.ts";
+import { API, APIError } from "../utils/api.ts";
+import { ManifestEntry } from "../utils/api_types.ts";
+import { parseEntrypoint } from "../utils/entrypoint.ts";
+
+const help = `deployctl deploy
+Deploy a script with static files to Deno Deploy.
+
+To deploy a local script:
+  deployctl deploy --project=helloworld main.ts
+
+To deploy a remote script:
+  deployctl deploy --project=helloworld https://deno.com/examples/hello.js
+
+To deploy a remote script without static files:
+  deployctl deploy --project=helloworld --no-static https://deno.com/examples/hello.js
+
+USAGE:
+    deployctl deploy [OPTIONS] <ENTRYPOINT>
+
+OPTIONS:
+        --exclude=<PATTERNS>  Exclude files that match this pattern
+        --include=<PATTERNS>  Only upload files that match this pattern
+    -h, --help                Prints help information
+        --no-static           Don't include the files in the CWD as static files
+        --prod                Create a production deployment (default is preview deployment)
+    -p, --project=NAME        The project to deploy to
+        --token=TOKEN         The API token to use (defaults to DEPLOY_TOKEN env var)
+`;
+
+export interface Args {
+  help: boolean;
+  static: boolean;
+  prod: boolean;
+  exclude?: string[];
+  include?: string[];
+  token: string | null;
+  project: string | null;
+}
+
+// deno-lint-ignore no-explicit-any
+export default async function (rawArgs: Record<string, any>): Promise<void> {
+  const args: Args = {
+    help: !!rawArgs.help,
+    static: !!rawArgs.static,
+    prod: !!rawArgs.prod,
+    token: rawArgs.token ? String(rawArgs.token) : null,
+    project: rawArgs.project ? String(rawArgs.project) : null,
+    exclude: rawArgs.exclude?.split(","),
+    include: rawArgs.include?.split(","),
+  };
+  const entrypoint: string | null = typeof rawArgs._[0] === "string"
+    ? rawArgs._[0]
+    : null;
+  if (args.help) {
+    console.log(help);
+    Deno.exit(0);
+  }
+  if (entrypoint === null) {
+    console.error(help);
+    error("No entrypoint specifier given.");
+  }
+  if (rawArgs._.length > 1) {
+    console.error(help);
+    error("Too many positional arguments given.");
+  }
+  if (args.project === null) {
+    console.error(help);
+    error("Missing project ID.");
+  }
+  const token = args.token ?? Deno.env.get("DEPLOY_TOKEN") ?? null;
+  if (token === null) {
+    console.error(help);
+    error("Missing access token. Set via --token or DEPLOY_TOKEN.");
```

Comment:
Nit: we should probably move the `missing token error` to the top, so that's the first thing they fix before anything else.